frontend/packages/console-app/src/components/cloud-shell/CloudShellTerminal.tsx

```diff
@@ -31,8 +41,43 @@ const resource = {
   },
 };
 
-const CloudShellTerminal: React.FC<CloudShellTerminalProps> = ({ username, onCancel }) => {
+const CloudShellTerminal: React.FC<CloudShellTerminalProps> = ({
+  username,
+  onCancel,
+  shouldResize = true,
+}) => {
   const [data, loaded, loadError] = useK8sWatchResource<CloudShellResource>(resource);
+  const [workSpacePod, setWorkspacePod] = React.useState<InitResponseObject>(null);
+  const [apiError, setApiError] = React.useState<string>(null);
+
+  React.useEffect(() => {
+    if (Array.isArray(data)) {
+      const workspace = data.find(
+        (d) => d?.metadata?.annotations?.[CLOUD_SHELL_USER_ANNOTATION] === username,
+      );
+
+      if (workspace) {
+        const running = workspace.status?.phase === 'Running';
+        if (!workSpacePod?.namespace) {
+          setWorkspacePod({ ...workSpacePod, namespace: workspace.metadata.namespace });
+        }
+        if (running && !apiError && !workSpacePod?.pod) {
+          makeTerminalInitCalls(username, workspace.metadata.name, workspace.metadata.namespace)
+            .then((res) => {
+              setWorkspacePod({
+                ...workSpacePod,
+                pod: res.pod,
+                container: res.container,
+                command: res.cmd || [],
+              });
+            })
+            .catch(() => {
+              setApiError('Failed to connect to your OpenShift command line terminal');
+            });
+        }
+      }
+    }
+  }, [apiError, data, username, workSpacePod]);
```

Comment:
`apiError` is set inside this effect and therefore shouldn't be a dependency. It will cause needless re-entry. You shouldn't need to check if it's set or not.

`workSpacePod` shouldn't be a dependency. It should be set only and not read. After making the call to `makeTerminalInitCalls` you set this value.

You could have it as a dependency but only to reset the value to `null` because we're trying to go through the initialization phase again.